File: Data Structures & Algorithms/is-palindrome/submission-8.py
class Solution:
    def isPalindrome(self, s: str) -> bool:

        # 建一個新 list
        # 去除空格
        # 大寫轉小寫
        # 用左右指針 不一樣就 return false 

        ans_s = []

        for char in s:
            if char.isalnum():
                ans_s.append(char.lower())

        left = 0
        right = len(ans_s) - 1

        while left < right:
            if ans_s[left] != ans_s[right]:
                return False
            left+= 1
            right-=1

        return True

        

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-8.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {

        int left = 0;
        int right = 1;
        int max_profit = 0;
        int n = prices.size();

        while (right < n)
        {
            if (prices[right] < prices[left])
            {
                left = right;
            } 
            else if (prices[right] > prices[left])
            {
                int curr_profit = prices[right] - prices[left];
                max_profit = max(curr_profit,max_profit);
            }
            right++;
        }

        return max_profit;
        
    }
};
